src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs

```diff
@@ -15,55 +15,186 @@ namespace Microsoft.Maui.Controls.BindingSourceGen;
+				{
+					codeWriter.AddBinding(binding);
+				}
 			}
+
+			spc.AddSource("GeneratedBindableObjectExtensions.g.cs", codeWriter.GenerateCode());
 		});
+	}
 
-		var endpoints = bindingsWithDiagnostics
-			.Where(endpoint => endpoint.Diagnostics.Count == 0)
-			.WithTrackingName(GeneratorSteps.EndpointsWithoutDiagnosicsStep);
+	static bool IsSetBindingMethod(SyntaxNode node)
+	{
+		return node is InvocationExpressionSyntax invocation
+			&& invocation.Expression is MemberAccessExpressionSyntax method
+			&& method.Name.Identifier.Text == "SetBinding";
+	}
 
-		context.RegisterSourceOutput(bindings, (context, bindings) =>
+	static bool IsNullable(ITypeSymbol type)
+	{
+		if (type.IsValueType)
 		{
-			using var codeWriter = new BindingCodeWriter();
-			foreach (var binding in bindings)
-			{
-				codeWriter.AddBinding(binding);
-			}
+			return false; // TODO: Fix
+		}
+		if (type.NullableAnnotation == NullableAnnotation.Annotated)
+		{
+			return true;
+		}
+		return false;
+	}
 
-			context.AddSource("GeneratedBindableObjectExtensions.g.cs", codeWriter.GenerateCode());
-		});
+	static Binding? GetBindingForGeneration(GeneratorSyntaxContext context, CancellationToken t)
+	{
+		var invocation = (InvocationExpressionSyntax)context.Node;
+
+		var method = invocation.Expression as MemberAccessExpressionSyntax;
+
+
+		var sourceCodeLocation = new SourceCodeLocation(
+			context.Node.SyntaxTree.FilePath,
+			method!.Name.GetLocation().GetLineSpan().StartLinePosition.Line + 1,
+			method!.Name.GetLocation().GetLineSpan().StartLinePosition.Character + 1
+		);
+
+		var argumentList = invocation.ArgumentList.Arguments;
+		var getter = argumentList[1].Expression;
+
+
+		if (getter is not LambdaExpressionSyntax lambda)
+		{
+			return null; // TODO: Optimize
+		}
+
+
+		if (context.SemanticModel.GetSymbolInfo(lambda).Symbol is not IMethodSymbol symbol)
+		{
+			return null;
+		}; // TODO
+
+		var inputType = symbol.Parameters[0].Type;
+		var inputTypeGlobalPath = inputType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+
+		var outputType = symbol.ReturnType;
+		var outputTypeGlobalPath = symbol.ReturnType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+
+		var inputTypeIsGenericParameter = inputType.Kind == SymbolKind.TypeParameter;
+		var outputTypeIsGenericParameter = outputType.Kind == SymbolKind.TypeParameter;
+
+		var parts = new List<PathPart>();
+		var types = new List<string>();
+		ParsePath(lambda.Body, context, parts, types);
+
+
+		// Convert to string
+		var pathasString = string.Join("<>", parts.Select(p => p.MemberName));
+		var typesAsString = string.Join("<>", types);
+
+
+		return new Binding(
+			Id: ++_idCounter,
+			Location: sourceCodeLocation,
+			SourceType: new TypeName(inputTypeGlobalPath, IsNullable(inputType), inputTypeIsGenericParameter),
+			PropertyType: new TypeName(outputTypeGlobalPath, IsNullable(outputType), outputTypeIsGenericParameter),
+			Path: [.. parts],
```

Comment:
I'd prefer using `parts.ToArray()` here to make it clearer that this copies the list into an array.